pkg/virt-controller/watch/migration.go

```diff
@@ -431,10 +474,143 @@ func (c *MigrationController) createTargetPod(migration *virtv1.VirtualMachineIn
 		c.podExpectations.CreationObserved(key)
 		return fmt.Errorf("failed to create vmi migration target pod: %v", err)
 	}
+	log.Log.Object(vmi).Infof("Created migration target pod %s/%s with uuid %s for migration %s with uuid %s", pod.Namespace, pod.Name, string(pod.UID), migration.Name, string(migration.UID))
 	c.recorder.Eventf(migration, k8sv1.EventTypeNormal, SuccessfulCreatePodReason, "Created migration target pod %s", pod.Name)
 	return nil
 }
 
+func (c *MigrationController) handleMarkMigrationFailedOnVMI(migration *virtv1.VirtualMachineInstanceMigration, vmi *virtv1.VirtualMachineInstance) error {
+
+	// Mark Migration Done on VMI if virt handler never started it.
+	// Once virt-handler starts the migration, it's up to handler
+	// to finalize it.
+
+	vmiCopy := vmi.DeepCopy()
+
+	now := v1.NewTime(time.Now())
+	vmiCopy.Status.MigrationState.StartTimestamp = &now
+	vmiCopy.Status.MigrationState.EndTimestamp = &now
+	vmiCopy.Status.MigrationState.Failed = true
+	vmiCopy.Status.MigrationState.Completed = true
+
+	err := c.patchVMI(vmi, vmiCopy)
+	if err != nil {
+		return err
```

Comment:
It *might* be worth a log message here. We could get an error from problems marshalling json, or problems patching the VMI. Arguably hard to do, but problems marshalling might be really hard to track down.

Reply:
agreed, fixed